dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -948,6 +948,7 @@ BlockInputStreamPtr Segment::getInputStream(
         expected_block_size,
         columns_to_read,
         segment_snap->stable->stable);
+    auto real_ranges = shrinkRowKeyRanges(read_ranges);
```

Comment:
Can we directly return an `EmptyBlockInputStream` when `real_ranges.empty()` here? And remove it from `getInputStreamModeNormal`/`getInputStreamModeFast`/`getInputStreamModeRaw`/`getBitmapFilterInputStream`
```
    if (real_ranges.empty())	
    {	
        return std::make_shared<EmptyBlockInputStream>(toEmptyBlock(columns_to_read));
    }
```